addons/controls/README.md

```diff
@@ -0,0 +1,230 @@
+
+- **Convenience.** Auto-generated based on your components (even for non-React frameworks)
+- **Compatibility.** Controls is based on [Storybook Args](https://github.com/storybookjs/storybook/blob/next/docs/src/pages/formats/component-story-format/index.md#args-story-inputs), so stories written for Controls will be portable:
+  - **Documentation.** Controls 100% compatible with [Storybook Docs](https://github.com/storybookjs/storybook/tree/next/addons/docs).
+  - **Testing.** Re-use stories directly inside testing tools like [Jest](https://jestjs.io/).
+  - **And more.** Re-use stories in design and development tools as Args is adopted.
+
+Most importantly, Controls incorporates lessons from years of supporting Knobs on tens of thousands of projects and dozens of different frameworks. We couldn't incrementally fix knobs, so we built a better version.
+
+- [Get started](#get-started)
+- [Installation](#installation)
+- [Writing stories](#writing-stories)
+  - [Auto-generated args](#auto-generated-args)
+  - [Manually-configured args](#manually-configured-args)
+- [Migrating from knobs](#migrating-from-knobs)
+  - [Knobs to auto-generated args](#knobs-to-auto-generated-args)
+  - [Knobs to manually-configured args](#knobs-to-manually-configured-args)
+- [Available controls](#available-controls)
+- [Parameters](#parameters)
+    - [expanded](#expanded)
+- [FAQs](#faqs)
+  - [How will this replace addon-knobs?](#how-will-this-replace-addon-knobs)
+
+## Get started
+
+To get started with `addon-controls`:
+
+1. [install the addon](#installation)
+2. [write an args story](#writing-stories)
+3. [configure the addon (optional)](#configuration)
+
+If you're installing it into an existing project that uses `addon-knobs`, you might also [migrate your existing stories](#migrating-from-knobs).
+
+## Installation
+
+In the first version, Controls requires that you're [Storybook Docs](https://github.com/storybookjs/storybook/tree/master/addons/docs). If you're not using it already, please install that first.
+
+Next, install the package:
+
+```sh
+npm install @storybook/addon-controls -D # or yarn
+```
+
+And add it to your `.storybook/main.js` config:
+
+```js
+module.exports = {
+  addons: ['@storybook/addon-controls'],
+};
+```
+
+Once the addon is installed you should see a `Controls` tab in the addons panel:
+
+## Writing stories
+
+Controls are only available for stories that make use of [Storybook Args](https://github.com/storybookjs/storybook/blob/next/docs/src/pages/formats/component-story-format/index.md#args-story-inputs), so if your story functions don't accept an `Args` object as the first argument, you'll need to upgrade your stories.
+
+Here's a `react` story that doesn't use args:
+
+```js
+export const Basic = () => <Button label="hello" />;
+```
+
+And here's one that does:
+
+```js
+export const Basic = (args = { label: 'hello' }) => <Button {...args} />;
```

Comment:
Does this work? I thought we didn't support this (default value) yet.

Reply:
Normally the storyFn would be called with the storyContext right? So the storyFn has access to parameters etc? Is this now the second argument sometimes?

Reply:
@ndelangen it's the second argument by default, unless you set the parameter `passArgsFirst: false`